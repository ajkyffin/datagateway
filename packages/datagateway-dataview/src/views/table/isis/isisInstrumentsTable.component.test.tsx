import React from 'react';
import { createMount } from '@material-ui/core/test-utils';
import ISISInstrumentsTable from './isisInstrumentsTable.component';
import { initialState as dgDataViewInitialState } from '../../../state/reducers/dgdataview.reducer';
import { StateType } from '../../../state/app.types';
import {
  // fetchInstrumentsRequest,
  // filterTable,
  // sortTable,
  // fetchInstrumentDetailsRequest,
  // fetchInstrumentCountRequest,
  Instrument,
  useInstrumentCount,
  useInstrumentsInfinite,
  dGCommonInitialState,
} from 'datagateway-common';
import { ReactWrapper } from 'enzyme';
import configureStore from 'redux-mock-store';
import { QueryClient, QueryClientProvider } from 'react-query';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { Router } from 'react-router';
import { createMemoryHistory, History } from 'history';
// import axios from 'axios';

jest.mock('datagateway-common', () => {
  const originalModule = jest.requireActual('datagateway-common');

  return {
    __esModule: true,
    ...originalModule,
    useInstrumentCount: jest.fn(),
    useInstrumentsInfinite: jest.fn(),
  };
});

describe('ISIS Instruments table component', () => {
  // let shallow;
  let mount;
  let mockStore;
  let state: StateType;
  let rowData: Instrument[];
  let history: History;

  const createWrapper = (studyHierarchy?: boolean): ReactWrapper => {
    const store = mockStore(state);
    return mount(
      <Provider store={store}>
        <Router history={history}>
          <QueryClientProvider client={new QueryClient()}>
            <ISISInstrumentsTable
              studyHierarchy={studyHierarchy ? true : false}
            />
          </QueryClientProvider>
        </Router>
      </Provider>
    );
  };

  beforeEach(() => {
    // shallow = createShallow({ untilSelector: 'ISISInstrumentsTable' });
    mount = createMount();
    rowData = [
      {
        id: 1,
        name: 'Test 1',
        fullName: 'Test instrument 1',
        description: 'foo bar',
        url: 'test url',
      },
      {
        id: 2,
        name: 'Test 2',
        description: 'foo bar',
        url: 'test url',
      },
    ];
    history = createMemoryHistory();

    mockStore = configureStore([thunk]);
    state = JSON.parse(
      JSON.stringify({
        dgdataview: dgDataViewInitialState,
        dgcommon: dGCommonInitialState,
      })
    );
    // state.dgcommon.data = [
    //   {
    //     id: 1,
    //     name: 'Test 1',
    //     fullName: 'Test instrument 1',
    //     description: 'foo bar',
    //     url: 'test url',
    //   },
    //   {
    //     id: 2,
    //     name: 'Test 2',
    //     description: 'foo bar',
    //     url: 'test url',
    //   },
    // ];

    (useInstrumentCount as jest.Mock).mockReturnValue(0);
    (useInstrumentsInfinite as jest.Mock).mockReturnValue({
      data: { pages: [rowData] },
      fetchNextPage: jest.fn(),
    });
  });

  afterEach(() => {
    mount.cleanUp();
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    // const wrapper = shallow(<ISISInstrumentsTable store={mockStore(state)} />);

    const wrapper = createWrapper();
    expect(wrapper.find('VirtualizedTable').props()).toMatchSnapshot();
  });

  // it('sends fetchInstrumentCount and fetchInstruments actions when watched store values change', () => {
  //   let testStore = mockStore(state);
  //   const wrapper = mount(
  //     <Provider store={testStore}>
  //       <MemoryRouter>
  //         <ISISInstrumentsTable />
  //       </MemoryRouter>
  //     </Provider>
  //   );

  //   // simulate clearTable action
  //   testStore = mockStore({
  //     ...state,
  //     dgdataview: { ...state.dgdataview, sort: {}, filters: {} },
  //   });
  //   wrapper.setProps({ store: testStore });

  //   expect(testStore.getActions()[0]).toEqual(fetchInstrumentCountRequest(1));
  //   expect(testStore.getActions()[1]).toEqual(fetchInstrumentsRequest(1));
  // });

  it('calls the correct data fetching hooks on load', () => {
    createWrapper();
    expect(useInstrumentCount).toHaveBeenCalled();
    expect(useInstrumentsInfinite).toHaveBeenCalled();
  });

  it('calls useInstrumentsInfinite when loadMoreRows is called', () => {
    // const testStore = mockStore(state);
    // const wrapper = shallow(<ISISInstrumentsTable store={testStore} />);

    // wrapper.prop('loadMoreRows')({ startIndex: 50, stopIndex: 74 });

    // expect(testStore.getActions()[0]).toEqual(fetchInstrumentsRequest(1));

    const fetchNextPage = jest.fn();
    (useInstrumentsInfinite as jest.Mock).mockReturnValueOnce({
      data: { pages: [rowData] },
      fetchNextPage,
    });
    const wrapper = createWrapper();

    wrapper.find('VirtualizedTable').prop('loadMoreRows')({
      startIndex: 50,
      stopIndex: 74,
    });

    expect(fetchNextPage).toHaveBeenCalledWith({
      pageParam: { startIndex: 50, stopIndex: 74 },
    });
  });

  it('sends filterTable action on filter', () => {
    // const testStore = mockStore(state);
    // const wrapper = mount(
    //   <Provider store={testStore}>
    //     <MemoryRouter>
    //       <ISISInstrumentsTable />
    //     </MemoryRouter>
    //   </Provider>
    // );

    // const filterInput = wrapper.find('input').first();
    // filterInput.instance().value = 'test';
    // filterInput.simulate('change');

    // expect(testStore.getActions()[2]).toEqual(
    //   filterTable('fullName', { value: 'test', type: 'include' })
    // );

    // filterInput.instance().value = '';
    // filterInput.simulate('change');

    // expect(testStore.getActions()[4]).toEqual(filterTable('fullName', null));

    const wrapper = createWrapper();

    const filterInput = wrapper.find('input').first();
    filterInput.instance().value = 'test';
    filterInput.simulate('change');

    expect(history.length).toBe(2);
    expect(history.location.search).toBe(
      `?filters=${encodeURIComponent(
        '{"fullName":{"value":"test","type":"include"}}'
      )}`
    );

    filterInput.instance().value = '';
    filterInput.simulate('change');

    expect(history.length).toBe(3);
    expect(history.location.search).toBe('?');
  });

  it('updates sort query params on sort', () => {
    // const testStore = mockStore(state);
    // const wrapper = mount(
    //   <Provider store={testStore}>
    //     <MemoryRouter>
    //       <ISISInstrumentsTable />
    //     </MemoryRouter>
    //   </Provider>
    // );

    // wrapper
    //   .find('[role="columnheader"] span[role="button"]')
    //   .first()
    //   .simulate('click');

    // expect(testStore.getActions()[2]).toEqual(sortTable('fullName', 'asc'));

    const wrapper = createWrapper();

    wrapper
      .find('[role="columnheader"] span[role="button"]')
      .first()
      .simulate('click');

    expect(history.length).toBe(2);
    expect(history.location.search).toBe(
      `?sort=${encodeURIComponent('{"fullName":"asc"}')}`
    );
  });

  // TODO: Come back to this as there is hooks error
  it.skip('renders details panel correctly and it calls useInstrumentDetails', () => {
    // const testStore = mockStore(state);
    // const wrapper = mount(
    //   <Provider store={testStore}>
    //     <MemoryRouter>
    //       <ISISInstrumentsTable />
    //     </MemoryRouter>
    //   </Provider>
    // );

    // const detailsPanelWrapper = shallow(
    //   wrapper.find(Table).prop('detailsPanel')({
    //     rowData: state.dgcommon.data[0],
    //   })
    // );
    // expect(detailsPanelWrapper).toMatchSnapshot();

    // mount(
    //   wrapper.find(Table).prop('detailsPanel')({
    //     rowData: state.dgcommon.data[0],
    //     detailsPanelResize: jest.fn(),
    //   })
    // );

    // expect(testStore.getActions()[2]).toEqual(fetchInstrumentDetailsRequest());

    const wrapper = createWrapper();

    const detailsPanelWrapper = mount(
      wrapper.find('VirtualizedTable').prop('detailsPanel')({
        rowData: rowData[0],
      })
    );
    expect(detailsPanelWrapper).toMatchSnapshot();
  });

  it('renders names as links', () => {
    // const wrapper = mount(
    //   <Provider store={mockStore(state)}>
    //     <MemoryRouter>
    //       <ISISInstrumentsTable />
    //     </MemoryRouter>
    //   </Provider>
    // );

    // expect(
    //   wrapper.find('[aria-colindex=2]').find('p').children()
    // ).toMatchSnapshot();

    const wrapper = createWrapper();

    expect(
      wrapper.find('[aria-colindex=2]').find('p').children()
    ).toMatchSnapshot();
  });

  it('renders names as links in StudyHierarchy', () => {
    // const wrapper = mount(
    //   <Provider store={mockStore(state)}>
    //     <MemoryRouter>
    //       <ISISInstrumentsTable studyHierarchy={true} />
    //     </MemoryRouter>
    //   </Provider>
    // );

    // expect(
    //   wrapper.find('[aria-colindex=2]').find('p').children()
    // ).toMatchSnapshot();

    const wrapper = createWrapper(true);

    expect(
      wrapper.find('[aria-colindex=2]').find('p').children()
    ).toMatchSnapshot();
  });
});
