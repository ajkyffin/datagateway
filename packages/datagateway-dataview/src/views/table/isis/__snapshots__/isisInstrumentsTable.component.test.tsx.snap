// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ISIS Instruments table component renders correctly 1`] = `
Object {
  "columns": Array [
    Object {
      "cellContentRenderer": [Function],
      "dataKey": "fullName",
      "defaultSort": "asc",
      "filterComponent": [Function],
      "icon": Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      "label": "instruments.name",
    },
    Object {
      "dataKey": "type",
      "filterComponent": [Function],
      "icon": Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      },
      "label": "instruments.type",
    },
  ],
  "data": Array [
    Object {
      "description": "foo bar",
      "fullName": "Test instrument 1",
      "id": 1,
      "name": "Test 1",
      "url": "test url",
    },
    Object {
      "description": "foo bar",
      "id": 2,
      "name": "Test 2",
      "url": "test url",
    },
  ],
  "detailsPanel": [Function],
  "loadMoreRows": [Function],
  "onSort": [Function],
  "sort": Object {
    "fullName": "asc",
  },
  "totalRowCount": 1,
}
`;

exports[`ISIS Instruments table component renders names as links in StudyHierarchy 1`] = `
Array [
  <ForwardRef(Link)
    component={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "propTypes": Object {
          "innerRef": [Function],
          "onClick": [Function],
          "replace": [Function],
          "target": [Function],
          "to": [Function],
        },
        "render": [Function],
      }
    }
    data-testid="isis-instrument-table-name"
    to="/browseStudyHierarchy/instrument/1/study"
  >
    <MuiLinkRoot
      className="MuiLink-root MuiLink-underlineAlways"
      color="primary"
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      data-testid="isis-instrument-table-name"
      onBlur={[Function]}
      onFocus={[Function]}
      ownerState={
        Object {
          "children": "Test instrument 1",
          "color": "primary",
          "component": Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          },
          "data-testid": "isis-instrument-table-name",
          "focusVisible": false,
          "to": "/browseStudyHierarchy/instrument/1/study",
          "underline": "always",
          "variant": "inherit",
        }
      }
      to="/browseStudyHierarchy/instrument/1/study"
      variant="inherit"
    >
      <Noop />
      <ForwardRef(Typography)
        className="MuiLink-root MuiLink-underlineAlways css-okiaqn-MuiLink-root"
        color="primary"
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        data-testid="isis-instrument-table-name"
        onBlur={[Function]}
        onFocus={[Function]}
        to="/browseStudyHierarchy/instrument/1/study"
        variant="inherit"
      >
        <MuiTypographyRoot
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "propTypes": Object {
                "innerRef": [Function],
                "onClick": [Function],
                "replace": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-okiaqn-MuiLink-root"
          data-testid="isis-instrument-table-name"
          onBlur={[Function]}
          onFocus={[Function]}
          ownerState={
            Object {
              "align": "inherit",
              "children": "Test instrument 1",
              "className": "MuiLink-root MuiLink-underlineAlways css-okiaqn-MuiLink-root",
              "color": "primary.main",
              "component": Object {
                "$$typeof": Symbol(react.forward_ref),
                "propTypes": Object {
                  "innerRef": [Function],
                  "onClick": [Function],
                  "replace": [Function],
                  "target": [Function],
                  "to": [Function],
                },
                "render": [Function],
              },
              "data-testid": "isis-instrument-table-name",
              "gutterBottom": false,
              "noWrap": false,
              "onBlur": [Function],
              "onFocus": [Function],
              "paragraph": false,
              "sx": Object {
                "color": "primary.main",
              },
              "to": "/browseStudyHierarchy/instrument/1/study",
              "variant": "inherit",
              "variantMapping": Object {
                "body1": "p",
                "body2": "p",
                "h1": "h1",
                "h2": "h2",
                "h3": "h3",
                "h4": "h4",
                "h5": "h5",
                "h6": "h6",
                "inherit": "p",
                "subtitle1": "h6",
                "subtitle2": "h6",
              },
            }
          }
          sx={
            Object {
              "color": "primary.main",
            }
          }
          to="/browseStudyHierarchy/instrument/1/study"
        >
          <Noop />
          <Link
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
            data-testid="isis-instrument-table-name"
            onBlur={[Function]}
            onFocus={[Function]}
            to="/browseStudyHierarchy/instrument/1/study"
          >
            <LinkAnchor
              className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
              data-testid="isis-instrument-table-name"
              href="/browseStudyHierarchy/instrument/1/study"
              navigate={[Function]}
              onBlur={[Function]}
              onFocus={[Function]}
            >
              <a
                className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
                data-testid="isis-instrument-table-name"
                href="/browseStudyHierarchy/instrument/1/study"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
              >
                Test instrument 1
              </a>
            </LinkAnchor>
          </Link>
        </MuiTypographyRoot>
      </ForwardRef(Typography)>
    </MuiLinkRoot>
  </ForwardRef(Link)>,
  <ForwardRef(Link)
    component={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "propTypes": Object {
          "innerRef": [Function],
          "onClick": [Function],
          "replace": [Function],
          "target": [Function],
          "to": [Function],
        },
        "render": [Function],
      }
    }
    data-testid="isis-instrument-table-name"
    to="/browseStudyHierarchy/instrument/2/study"
  >
    <MuiLinkRoot
      className="MuiLink-root MuiLink-underlineAlways"
      color="primary"
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      data-testid="isis-instrument-table-name"
      onBlur={[Function]}
      onFocus={[Function]}
      ownerState={
        Object {
          "children": "Test 2",
          "color": "primary",
          "component": Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          },
          "data-testid": "isis-instrument-table-name",
          "focusVisible": false,
          "to": "/browseStudyHierarchy/instrument/2/study",
          "underline": "always",
          "variant": "inherit",
        }
      }
      to="/browseStudyHierarchy/instrument/2/study"
      variant="inherit"
    >
      <Noop />
      <ForwardRef(Typography)
        className="MuiLink-root MuiLink-underlineAlways css-okiaqn-MuiLink-root"
        color="primary"
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        data-testid="isis-instrument-table-name"
        onBlur={[Function]}
        onFocus={[Function]}
        to="/browseStudyHierarchy/instrument/2/study"
        variant="inherit"
      >
        <MuiTypographyRoot
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "propTypes": Object {
                "innerRef": [Function],
                "onClick": [Function],
                "replace": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-okiaqn-MuiLink-root"
          data-testid="isis-instrument-table-name"
          onBlur={[Function]}
          onFocus={[Function]}
          ownerState={
            Object {
              "align": "inherit",
              "children": "Test 2",
              "className": "MuiLink-root MuiLink-underlineAlways css-okiaqn-MuiLink-root",
              "color": "primary.main",
              "component": Object {
                "$$typeof": Symbol(react.forward_ref),
                "propTypes": Object {
                  "innerRef": [Function],
                  "onClick": [Function],
                  "replace": [Function],
                  "target": [Function],
                  "to": [Function],
                },
                "render": [Function],
              },
              "data-testid": "isis-instrument-table-name",
              "gutterBottom": false,
              "noWrap": false,
              "onBlur": [Function],
              "onFocus": [Function],
              "paragraph": false,
              "sx": Object {
                "color": "primary.main",
              },
              "to": "/browseStudyHierarchy/instrument/2/study",
              "variant": "inherit",
              "variantMapping": Object {
                "body1": "p",
                "body2": "p",
                "h1": "h1",
                "h2": "h2",
                "h3": "h3",
                "h4": "h4",
                "h5": "h5",
                "h6": "h6",
                "inherit": "p",
                "subtitle1": "h6",
                "subtitle2": "h6",
              },
            }
          }
          sx={
            Object {
              "color": "primary.main",
            }
          }
          to="/browseStudyHierarchy/instrument/2/study"
        >
          <Noop />
          <Link
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
            data-testid="isis-instrument-table-name"
            onBlur={[Function]}
            onFocus={[Function]}
            to="/browseStudyHierarchy/instrument/2/study"
          >
            <LinkAnchor
              className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
              data-testid="isis-instrument-table-name"
              href="/browseStudyHierarchy/instrument/2/study"
              navigate={[Function]}
              onBlur={[Function]}
              onFocus={[Function]}
            >
              <a
                className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
                data-testid="isis-instrument-table-name"
                href="/browseStudyHierarchy/instrument/2/study"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
              >
                Test 2
              </a>
            </LinkAnchor>
          </Link>
        </MuiTypographyRoot>
      </ForwardRef(Typography)>
    </MuiLinkRoot>
  </ForwardRef(Link)>,
]
`;

exports[`ISIS Instruments table component renders names as links when NOT in studyHierarchy 1`] = `
Array [
  <ForwardRef(Link)
    component={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "propTypes": Object {
          "innerRef": [Function],
          "onClick": [Function],
          "replace": [Function],
          "target": [Function],
          "to": [Function],
        },
        "render": [Function],
      }
    }
    data-testid="isis-instrument-table-name"
    to="/browse/instrument/1/facilityCycle"
  >
    <MuiLinkRoot
      className="MuiLink-root MuiLink-underlineAlways"
      color="primary"
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      data-testid="isis-instrument-table-name"
      onBlur={[Function]}
      onFocus={[Function]}
      ownerState={
        Object {
          "children": "Test instrument 1",
          "color": "primary",
          "component": Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          },
          "data-testid": "isis-instrument-table-name",
          "focusVisible": false,
          "to": "/browse/instrument/1/facilityCycle",
          "underline": "always",
          "variant": "inherit",
        }
      }
      to="/browse/instrument/1/facilityCycle"
      variant="inherit"
    >
      <Noop />
      <ForwardRef(Typography)
        className="MuiLink-root MuiLink-underlineAlways css-okiaqn-MuiLink-root"
        color="primary"
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        data-testid="isis-instrument-table-name"
        onBlur={[Function]}
        onFocus={[Function]}
        to="/browse/instrument/1/facilityCycle"
        variant="inherit"
      >
        <MuiTypographyRoot
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "propTypes": Object {
                "innerRef": [Function],
                "onClick": [Function],
                "replace": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-okiaqn-MuiLink-root"
          data-testid="isis-instrument-table-name"
          onBlur={[Function]}
          onFocus={[Function]}
          ownerState={
            Object {
              "align": "inherit",
              "children": "Test instrument 1",
              "className": "MuiLink-root MuiLink-underlineAlways css-okiaqn-MuiLink-root",
              "color": "primary.main",
              "component": Object {
                "$$typeof": Symbol(react.forward_ref),
                "propTypes": Object {
                  "innerRef": [Function],
                  "onClick": [Function],
                  "replace": [Function],
                  "target": [Function],
                  "to": [Function],
                },
                "render": [Function],
              },
              "data-testid": "isis-instrument-table-name",
              "gutterBottom": false,
              "noWrap": false,
              "onBlur": [Function],
              "onFocus": [Function],
              "paragraph": false,
              "sx": Object {
                "color": "primary.main",
              },
              "to": "/browse/instrument/1/facilityCycle",
              "variant": "inherit",
              "variantMapping": Object {
                "body1": "p",
                "body2": "p",
                "h1": "h1",
                "h2": "h2",
                "h3": "h3",
                "h4": "h4",
                "h5": "h5",
                "h6": "h6",
                "inherit": "p",
                "subtitle1": "h6",
                "subtitle2": "h6",
              },
            }
          }
          sx={
            Object {
              "color": "primary.main",
            }
          }
          to="/browse/instrument/1/facilityCycle"
        >
          <Noop />
          <Link
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
            data-testid="isis-instrument-table-name"
            onBlur={[Function]}
            onFocus={[Function]}
            to="/browse/instrument/1/facilityCycle"
          >
            <LinkAnchor
              className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
              data-testid="isis-instrument-table-name"
              href="/browse/instrument/1/facilityCycle"
              navigate={[Function]}
              onBlur={[Function]}
              onFocus={[Function]}
            >
              <a
                className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
                data-testid="isis-instrument-table-name"
                href="/browse/instrument/1/facilityCycle"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
              >
                Test instrument 1
              </a>
            </LinkAnchor>
          </Link>
        </MuiTypographyRoot>
      </ForwardRef(Typography)>
    </MuiLinkRoot>
  </ForwardRef(Link)>,
  <ForwardRef(Link)
    component={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "propTypes": Object {
          "innerRef": [Function],
          "onClick": [Function],
          "replace": [Function],
          "target": [Function],
          "to": [Function],
        },
        "render": [Function],
      }
    }
    data-testid="isis-instrument-table-name"
    to="/browse/instrument/2/facilityCycle"
  >
    <MuiLinkRoot
      className="MuiLink-root MuiLink-underlineAlways"
      color="primary"
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      data-testid="isis-instrument-table-name"
      onBlur={[Function]}
      onFocus={[Function]}
      ownerState={
        Object {
          "children": "Test 2",
          "color": "primary",
          "component": Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          },
          "data-testid": "isis-instrument-table-name",
          "focusVisible": false,
          "to": "/browse/instrument/2/facilityCycle",
          "underline": "always",
          "variant": "inherit",
        }
      }
      to="/browse/instrument/2/facilityCycle"
      variant="inherit"
    >
      <Noop />
      <ForwardRef(Typography)
        className="MuiLink-root MuiLink-underlineAlways css-okiaqn-MuiLink-root"
        color="primary"
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        data-testid="isis-instrument-table-name"
        onBlur={[Function]}
        onFocus={[Function]}
        to="/browse/instrument/2/facilityCycle"
        variant="inherit"
      >
        <MuiTypographyRoot
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "propTypes": Object {
                "innerRef": [Function],
                "onClick": [Function],
                "replace": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-okiaqn-MuiLink-root"
          data-testid="isis-instrument-table-name"
          onBlur={[Function]}
          onFocus={[Function]}
          ownerState={
            Object {
              "align": "inherit",
              "children": "Test 2",
              "className": "MuiLink-root MuiLink-underlineAlways css-okiaqn-MuiLink-root",
              "color": "primary.main",
              "component": Object {
                "$$typeof": Symbol(react.forward_ref),
                "propTypes": Object {
                  "innerRef": [Function],
                  "onClick": [Function],
                  "replace": [Function],
                  "target": [Function],
                  "to": [Function],
                },
                "render": [Function],
              },
              "data-testid": "isis-instrument-table-name",
              "gutterBottom": false,
              "noWrap": false,
              "onBlur": [Function],
              "onFocus": [Function],
              "paragraph": false,
              "sx": Object {
                "color": "primary.main",
              },
              "to": "/browse/instrument/2/facilityCycle",
              "variant": "inherit",
              "variantMapping": Object {
                "body1": "p",
                "body2": "p",
                "h1": "h1",
                "h2": "h2",
                "h3": "h3",
                "h4": "h4",
                "h5": "h5",
                "h6": "h6",
                "inherit": "p",
                "subtitle1": "h6",
                "subtitle2": "h6",
              },
            }
          }
          sx={
            Object {
              "color": "primary.main",
            }
          }
          to="/browse/instrument/2/facilityCycle"
        >
          <Noop />
          <Link
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
            data-testid="isis-instrument-table-name"
            onBlur={[Function]}
            onFocus={[Function]}
            to="/browse/instrument/2/facilityCycle"
          >
            <LinkAnchor
              className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
              data-testid="isis-instrument-table-name"
              href="/browse/instrument/2/facilityCycle"
              navigate={[Function]}
              onBlur={[Function]}
              onFocus={[Function]}
            >
              <a
                className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
                data-testid="isis-instrument-table-name"
                href="/browse/instrument/2/facilityCycle"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
              >
                Test 2
              </a>
            </LinkAnchor>
          </Link>
        </MuiTypographyRoot>
      </ForwardRef(Typography)>
    </MuiLinkRoot>
  </ForwardRef(Link)>,
]
`;
